orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/TitusInterestingHealthProviderNamesSupplier.java

```diff
@@ -0,0 +1,51 @@
+package com.netflix.spinnaker.orca.clouddriver.tasks.servergroup;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.spinnaker.orca.clouddriver.OortService;
+import com.netflix.spinnaker.orca.kato.pipeline.support.StageData;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import retrofit.client.Response;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Arrays;
+
+@Slf4j
+@Component
+public class TitusInterestingHealthProviderNamesSupplier implements InterestingHealthProviderNamesSupplier {
+  private static final String TITUS = "Titus";
+  private static final String INTERESTING_HEALTH_PROVIDER_NAMES = "interestingHealthProviderNames";
+  private static final String ROLLING_PUSH = "rollingpush";
+  private static final List<String> SUPPORTED_STAGES = Arrays.asList("deploy", "enableServerGroup", "createServerGroup", "cloneServerGroup", "disableServerGroup");
+
+  @Override
+  public boolean supports(String cloudProvider, Stage stage) {
+    return cloudProvider.equalsIgnoreCase(TITUS) && SUPPORTED_STAGES.contains(stage.getType()) && stage.getContext().containsKey("strategy") && stage.getContext().get("strategy").equals(ROLLING_PUSH);
+  }
+
+  @Override
+  public List<String> process(String cloudProvider, OortService oortService, ObjectMapper objectMapper, Stage stage) {
+    String region = (String) stage.getContext().get("region");
+    String asgName = (String) stage.getContext().get("asgName");
```

Comment:
Typically we would check both `serverGroupName` and `asgName` (former is _more_ correct but later is legacy and still commonly used).

Also, I believe that `asgName` / `serverGroupName` is contained within `stage.context.source`. See `SourceResolver.getSource(stage)` that `RollingPushStrategy` is using.

`deck` is sending the following payload for a rolling push:

```
{
    "job": [{
        "application": "spintest",
        "strategy": "rollingpush",
        "stack": "",
        "freeFormDetails": "",
        "credentials": "test",
        "cooldown": 10,
        "healthCheckGracePeriod": 600,
        "healthCheckType": "EC2",
        "terminationPolicies": ["Default"],
        "loadBalancers": [],
        "useSourceCapacity": true,
        "capacity": {
            "min": 1,
            "max": 1,
            "desired": 1
        },
        "targetHealthyDeployPercentage": 100,
        "availabilityZones": {
            "us-east-1": ["us-east-1c", "us-east-1d", "us-east-1e"]
        },
        "source": {
            "account": "test",
            "region": "us-east-1",
            "asgName": "spintest-v002"
        },
        "suspendedProcesses": [],
        "tags": {},
        "useAmiBlockDeviceMappings": false,
        "subnetType": "internal (vpc0)",
        "instanceType": "t2.micro",
        "iamRole": "........",
        "keyPair": "........",
        "instanceMonitoring": false,
        "ebsOptimized": false,
        "base64UserData": null,
        "securityGroups": [........],
        "amiName": "........I",
        "virtualizationType": "hvm",
        "termination": {
            "order": "oldest",
            "relaunchAllInstances": true,
            "concurrentRelaunches": 1,
            "totalRelaunches": 1
        },
        "type": "cloneServerGroup",
        "cloudProvider": "aws",
        "account": "test",
        "user": "ajordens@netflix.com"
    }],
    "application": "spintest",
    "description": "Create Cloned Server Group from spintest-v002"
}
```